### PR TITLE
Symlink includes

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/ncurses-feedstock/issues/3
Closes https://github.com/conda-forge/ncurses-feedstock/pull/4

Different Linux variants handle the issue of where to place ncurses differently. Debian apparently puts them all in `include` in a flat manner (no subdirectories). Enterprise Linux appears to also put them in a flat manner, but symlinks the headers from their usual locations `include/ncurses` and `include/ncursesw` to the flat structure. This seems to be the best of both worlds. People expecting to do `#include <ncurses/curses.h>` for instances can still do this with this strategy, but it also allows for things like `#include <curses.h>`. That way applications expecting to find them in `${PREFIX}/include/ncursues` and `${PREFIX}/include/ncursuesw` can find them, but they are also available in `${PREFIX}/include` for those that prefer to find them there. Note that symlinking widec and and non-widec includes to the same includes is ok because they are [source compatible]( http://www.linuxfromscratch.org/lfs/view/6.5/chapter06/ncurses.html ).

I have absorbed PR ( https://github.com/conda-forge/ncurses-feedstock/pull/4 ) into the history here, but have changed how it works to better solve this problem as explained above. Feedback welcome. Thanks @jjhelmus for raising this point.

cc @pelson @jjhelmus

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -15,8 +19,22 @@ do
...
+    HEADERS_DIR="${PREFIX}/include/ncurses"
+    if [ "${USE_WIDEC}" = true ];
+    then
+        HEADERS_DIR="${PREFIX}/include/ncursesw"
+    fi
+    for HEADER in $(ls $HEADERS_DIR);
+    do
+        mv "${HEADERS_DIR}/${HEADER}" "${PREFIX}/include/${HEADER}"
+        ln -s "${PREFIX}/include/${HEADER}" "${HEADERS_DIR}/${HEADER}"
+    done
```
Comment (jakirkham) [inline recipe/build.sh]: This is how we handle moving and symlinking. We don't have to worry about clobbering files as the headers are the exact same in all cases.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -16,25 +16,67 @@ source:
...
+    ] %}
+    {% for each_ncurses_library in ncurses_libraries %}
+    - test -f ${PREFIX}/lib/{{ each_ncurses_library }}.a
+    - test -f ${PREFIX}/lib/{{ each_ncurses_library }}.dylib   # [osx]
+    - test -f ${PREFIX}/lib/{{ each_ncurses_library }}.so      # [linux]
+    {% endfor %}
 
+    # Test include directories
+    - test -d ${PREFIX}/include/ncurses
+    - test -d ${PREFIX}/include/ncursesw
```
Comment (jakirkham) [inline recipe/meta.yaml]: Verify the subdirectories are present.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -16,25 +16,67 @@ source:
...
+        "term_entry.h",
+        "termcap.h",
+        "tic.h",
+        "unctrl.h"
+    ] %}
+    {% for each_ncurses_header in ncurses_headers %}
+    - test -L ${PREFIX}/include/ncurses/{{ each_ncurses_header }}
+    - test -L ${PREFIX}/include/ncursesw/{{ each_ncurses_header }}
+    - test -f ${PREFIX}/include/{{ each_ncurses_header }}
+    {% endfor %}
```
Comment (jakirkham) [inline recipe/meta.yaml]: Verify all the headers are present and that symlinks are used for the headers in the subdirectories.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -46,3 +88,4 @@ about:
 extra:
   recipe-maintainers:
     - jakirkham
+    - jjhelmus
```
Comment (jakirkham) [inline recipe/meta.yaml]: As I absorbed your PR, @jjhelmus, you are added as a maintainer. Hope this is still ok. Please let me know either way.

**Comment by jakirkham** [comment]
Added some tests to verify that all headers are in place. Also, added some tests for shared objects and dynamic libraries as they were not present before.

**Comment by jjhelmus** [comment]
LGTM. Happy to be added as a maintainer.

**Comment by jakirkham** [comment]
Alright, let's try this out and if we have more issues we can try something else. Just for my own curiosity, was there some particular kind of software this change was needed for?

**Comment by jjhelmus** [comment]
Python 3.5 needs ncurses and when I switched from the yum installed package I needed to add the new include path which seems a bit odd, conda-forge/staged-recipes#370.

**Comment by jakirkham** [comment]
Make sense. Thanks for explaining.

Yeah, this should be more inline with CentOS 6 so it should work nicely. Also, I added both widec and non-widec support earlier, which CentOS also does.

**Pull request merged**